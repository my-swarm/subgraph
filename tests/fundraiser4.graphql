query {
  fundraiser(id: "0x2fc631e4b3018258759c52af169200213e84abab") {
    id
    address
    owner
    token { id }
    label
    startDate
    endDate
    createdAt
    softCap
    hardCap
    supply
    baseCurrency {
      id
      address
      name
      symbol
      decimals
    }
    tokenPrice
    affiliateManager { address }
    contributorRestrictions
    fundraiserManager
    minter
    contributionsLocked

    amountQualified
    amountPending
    amountRefunded
    amountWithdrawn
    status
    numContributors
    contributors {
      id
      status
      address
      amount
      contributions {
        id
        type
        timestamp
        amount
      }
    }
    affiliates {
      id
      address
      referral
      percentage
      affiliateManager { address }
    }
    search
  }
}
