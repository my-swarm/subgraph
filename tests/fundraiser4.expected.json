{
  "data": {
    "fundraiser": {
      "address": "0x2fc631e4b3018258759c52af169200213e84abab",
      "affiliateManager": {
        "address": "0xd871ba711541027d031e4846169439ff959193da"
      },
      "affiliates": [
        {
          "address": "0x976ea74026e726554db657fa54763abd0c3a0aa9",
          "affiliateManager": {
            "address": "0xd871ba711541027d031e4846169439ff959193da"
          },
          "id": "0xd871ba711541027d031e4846169439ff959193da_0x976ea74026e726554db657fa54763abd0c3a0aa9",
          "percentage": "20000",
          "referral": "referral2"
        },
        {
          "address": "0x9965507d1a55bcc2695c58ba16fb37d819b0a4dc",
          "affiliateManager": {
            "address": "0xd871ba711541027d031e4846169439ff959193da"
          },
          "id": "0xd871ba711541027d031e4846169439ff959193da_0x9965507d1a55bcc2695c58ba16fb37d819b0a4dc",
          "percentage": "100000",
          "referral": "referral1"
        }
      ],
      "amountPending": "2000000000",
      "amountQualified": "5100000000",
      "amountRefunded": "1900000000",
      "amountWithdrawn": "0",
      "baseCurrency": {
        "address": "0xcf7ed3acca5a467e9e704c703e8d87f634fb0fc9",
        "decimals": 6,
        "id": "0xcf7ed3acca5a467e9e704c703e8d87f634fb0fc9",
        "name": "Mock USDC",
        "symbol": "USDC"
      },
      "contributionsLocked": false,
      "contributorRestrictions": "0x8edfefe2fbec2f521339b999ba4b09c19ee5e24d",
      "contributors": [
        {
          "address": "0x14dc79964da2c08b23698b3d3cc7ca32193d9955",
          "amount": "0",
          "contributions": [
            {
              "amount": "500000000",
              "id": "0x1b09d473ebfc10f650e895d4b5d72843e2bd9838790071b723d724380dfce521-0",
              "timestamp": 1616055522,
              "type": "Qualified"
            },
            {
              "amount": "500000000",
              "id": "0xc4dd12582f18f0a6d9c0adc2084bb84b9a8fc0518c9313717f5c3a9ece0541bc-2",
              "timestamp": 1616314737,
              "type": "Refunded"
            },
            {
              "amount": "500000000",
              "id": "0xcf2489b632a7f321250b846e0c03594d12ee4f6d32ae78599b7901c25e7af9ef-2",
              "timestamp": 1615969114,
              "type": "Pending"
            }
          ],
          "id": "0x2fc631e4b3018258759c52af169200213e84abab_0x14dc79964da2c08b23698b3d3cc7ca32193d9955",
          "status": "Refunded"
        },
        {
          "address": "0x15d34aaf54267db7d7c367839aaf71a00a2c6a65",
          "amount": "200000000",
          "contributions": [
            {
              "amount": "200000000",
              "id": "0x73c4ddb867b6151229218eb5639304024cabe4b2a69fd350f37808dac45bc53b-0",
              "timestamp": 1615969115,
              "type": "Qualified"
            },
            {
              "amount": "200000000",
              "id": "0xa04bf404d8a44e57ba29aed5f431bcc55c25fc1f5243628f9ad9119f4cbeb737-2",
              "timestamp": 1615882708,
              "type": "Pending"
            }
          ],
          "id": "0x2fc631e4b3018258759c52af169200213e84abab_0x15d34aaf54267db7d7c367839aaf71a00a2c6a65",
          "status": "Qualified"
        },
        {
          "address": "0x1cbd3b2770909d4e10f157cabc84c7264073c9ec",
          "amount": "500000000",
          "contributions": [
            {
              "amount": "500000000",
              "id": "0x5a537c19da0d4e45e026e9af34246f067fb1b535790810388b39346e6f2f9a5e-2",
              "timestamp": 1616228333,
              "type": "Pending"
            }
          ],
          "id": "0x2fc631e4b3018258759c52af169200213e84abab_0x1cbd3b2770909d4e10f157cabc84c7264073c9ec",
          "status": "Pending"
        },
        {
          "address": "0x23618e81e3f5cdf7f54c3d65f7fbc0abf5b21e8f",
          "amount": "0",
          "contributions": [
            {
              "amount": "400000000",
              "id": "0xefc2fd25591d2e8fd246896c787fc33992c1c43526fc0850b69b52fb4dbad777-2",
              "timestamp": 1616055519,
              "type": "Pending"
            },
            {
              "amount": "400000000",
              "id": "0xf5dba299dc87c7c66eacb9fb80b9c97f379544ce62a59da6d6dc4677c907217d-2",
              "timestamp": 1616055524,
              "type": "Removed"
            }
          ],
          "id": "0x2fc631e4b3018258759c52af169200213e84abab_0x23618e81e3f5cdf7f54c3d65f7fbc0abf5b21e8f",
          "status": "Qualified"
        },
        {
          "address": "0x2546bcd3c84621e976d8185a91a922ae77ecec30",
          "amount": "800000000",
          "contributions": [
            {
              "amount": "800000000",
              "id": "0x1f76c47504164c978ad90c63c717c45f3d27da7cdc0964e6108824767e28cd61-2",
              "timestamp": 1616573949,
              "type": "Qualified"
            }
          ],
          "id": "0x2fc631e4b3018258759c52af169200213e84abab_0x2546bcd3c84621e976d8185a91a922ae77ecec30",
          "status": "Qualified"
        },
        {
          "address": "0x71be63f3384f5fb98995898a86b02fb2426c5788",
          "amount": "500000000",
          "contributions": [
            {
              "amount": "500000000",
              "id": "0x4666f2b57be737ebcabde15d96401c276f38c9a36b5d7217306b746f32bbc898-0",
              "timestamp": 1616228335,
              "type": "Qualified"
            },
            {
              "amount": "500000000",
              "id": "0x61a7d7eb5f242e78d52f745b9e5cfd78c93a6799bb4d8f8ade26c8a3e03d36d8-2",
              "timestamp": 1616141928,
              "type": "Pending"
            }
          ],
          "id": "0x2fc631e4b3018258759c52af169200213e84abab_0x71be63f3384f5fb98995898a86b02fb2426c5788",
          "status": "Qualified"
        },
        {
          "address": "0x8626f6940e2eb28930efb4cef49b2d1f2c9c1199",
          "amount": "700000000",
          "contributions": [
            {
              "amount": "700000000",
              "id": "0xa0b7297922ad7fcc97757ee4977e3fc2bcb0aa87ec10e1f336c646aca4f7a28c-2",
              "timestamp": 1617005955,
              "type": "Qualified"
            }
          ],
          "id": "0x2fc631e4b3018258759c52af169200213e84abab_0x8626f6940e2eb28930efb4cef49b2d1f2c9c1199",
          "status": "Qualified"
        },
        {
          "address": "0x976ea74026e726554db657fa54763abd0c3a0aa9",
          "amount": "300000000",
          "contributions": [
            {
              "amount": "300000000",
              "id": "0x2f46e40f72a25fd366705cc6719da7253c69fab5c2c2be4a52327df934c53c3b-2",
              "timestamp": 1615969112,
              "type": "Pending"
            },
            {
              "amount": "300000000",
              "id": "0xee942a2eec96248f9f7289e1cd1c78b28e80441386fbd0247711f1bc47b4de68-0",
              "timestamp": 1615969117,
              "type": "Qualified"
            }
          ],
          "id": "0x2fc631e4b3018258759c52af169200213e84abab_0x976ea74026e726554db657fa54763abd0c3a0aa9",
          "status": "Qualified"
        },
        {
          "address": "0x9965507d1a55bcc2695c58ba16fb37d819b0a4dc",
          "amount": "100000000",
          "contributions": [
            {
              "amount": "100000000",
              "id": "0xd68841c80c98d784148170403a437b0bbe888eb86c1eace86613faec63019b7f-0",
              "timestamp": 1615969116,
              "type": "Qualified"
            },
            {
              "amount": "100000000",
              "id": "0xe88ab5e384495bb9ccce95ebbfd046c3c95630cc644f42a954f2d941363967d4-2",
              "timestamp": 1615882710,
              "type": "Pending"
            }
          ],
          "id": "0x2fc631e4b3018258759c52af169200213e84abab_0x9965507d1a55bcc2695c58ba16fb37d819b0a4dc",
          "status": "Qualified"
        },
        {
          "address": "0xa0ee7a142d267c1f36714e4a8f75612f20a79720",
          "amount": "0",
          "contributions": [
            {
              "amount": "500000000",
              "id": "0x21404f9438d413684c34c4dad04a9ef44959670b1229ed946d268377ed725cb6-2",
              "timestamp": 1616055521,
              "type": "Pending"
            },
            {
              "amount": "500000000",
              "id": "0x2cfd150809d461b05ae22f300f82435190f9b790119871be827f6248e233409c-0",
              "timestamp": 1616055523,
              "type": "Qualified"
            },
            {
              "amount": "500000000",
              "id": "0x31cd170f73259a4fd89fdd6ef8243d76e2e6ca12001c000b239f19907230bec8-2",
              "timestamp": 1616141929,
              "type": "Removed"
            }
          ],
          "id": "0x2fc631e4b3018258759c52af169200213e84abab_0xa0ee7a142d267c1f36714e4a8f75612f20a79720",
          "status": "Removed"
        },
        {
          "address": "0xbcd4042de499d14e55001ccbb24a551f3b954096",
          "amount": "0",
          "contributions": [
            {
              "amount": "500000000",
              "id": "0x03d1a0220a2b7c7c1b34201c7f1a3636ecf829d83131be4f88780d5e7019f879-2",
              "timestamp": 1616314736,
              "type": "Removed"
            },
            {
              "amount": "500000000",
              "id": "0xdc2bd184ad9396ce73b152eaf9e51dbba1afe1791abaa0680648e0d949630f5f-2",
              "timestamp": 1616141926,
              "type": "Pending"
            }
          ],
          "id": "0x2fc631e4b3018258759c52af169200213e84abab_0xbcd4042de499d14e55001ccbb24a551f3b954096",
          "status": "Removed"
        },
        {
          "address": "0xbda5747bfd65f08deb54cb465eb87d40e51b197e",
          "amount": "700000000",
          "contributions": [
            {
              "amount": "700000000",
              "id": "0xfa5249593a683ff1fa50c1b4c3322a0cd4e55648a332b58373f07fe2e0d25662-2",
              "timestamp": 1616746751,
              "type": "Qualified"
            }
          ],
          "id": "0x2fc631e4b3018258759c52af169200213e84abab_0xbda5747bfd65f08deb54cb465eb87d40e51b197e",
          "status": "Qualified"
        },
        {
          "address": "0xcd3b766ccdd6ae721141f452c550ca635964ce71",
          "amount": "400000000",
          "contributions": [
            {
              "amount": "400000000",
              "id": "0xf22838491cf1f4df943ef7569193dd38287748021b4481bc0858d2490808015b-2",
              "timestamp": 1616487547,
              "type": "Qualified"
            }
          ],
          "id": "0x2fc631e4b3018258759c52af169200213e84abab_0xcd3b766ccdd6ae721141f452c550ca635964ce71",
          "status": "Qualified"
        },
        {
          "address": "0xdd2fd4581271e230360230f9337d5c0430bf44c0",
          "amount": "700000000",
          "contributions": [
            {
              "amount": "300000000",
              "id": "0xecc894605adff25f6a85acf3958f04980c0b00d9c358a1f522d5399dcce0c8a3-2",
              "timestamp": 1617351559,
              "type": "Qualified"
            },
            {
              "amount": "400000000",
              "id": "0xf10fc73f8491ac906790c56c63a8415ee33ef4f2fa652071894a3456bfe24bf2-2",
              "timestamp": 1617005953,
              "type": "Qualified"
            }
          ],
          "id": "0x2fc631e4b3018258759c52af169200213e84abab_0xdd2fd4581271e230360230f9337d5c0430bf44c0",
          "status": "Qualified"
        },
        {
          "address": "0xdf3e18d64bc6a983f673ab319ccae4f1a57c7097",
          "amount": "700000000",
          "contributions": [
            {
              "amount": "700000000",
              "id": "0x2942122f2afdebb8081d9fd04e1bd2eac5e46dd51bf217f21937fa55d7e35d97-2",
              "timestamp": 1616487545,
              "type": "Qualified"
            }
          ],
          "id": "0x2fc631e4b3018258759c52af169200213e84abab_0xdf3e18d64bc6a983f673ab319ccae4f1a57c7097",
          "status": "Qualified"
        },
        {
          "address": "0xfabb0ac9d68b0b445fb7357272ff202c5651694a",
          "amount": "1500000000",
          "contributions": [
            {
              "amount": "500000000",
              "id": "0x44693d4ed25dc685eb31bfe15e8b99224ad90ec2a2c25c6e02a763d5391382af-2",
              "timestamp": 1616228331,
              "type": "Pending"
            },
            {
              "amount": "1000000000",
              "id": "0xbb338f4e8719d2292db626192387604d965d5d2cacfdfc320e854eab2b2ca9d7-2",
              "timestamp": 1617092357,
              "type": "Pending"
            }
          ],
          "id": "0x2fc631e4b3018258759c52af169200213e84abab_0xfabb0ac9d68b0b445fb7357272ff202c5651694a",
          "status": "Pending"
        }
      ],
      "createdAt": 1615882704,
      "endDate": 1642321036,
      "fundraiserManager": "0x8a791620dd6260079bf849dc5567adc3f2fdc318",
      "hardCap": "10000000000",
      "id": "0x2fc631e4b3018258759c52af169200213e84abab",
      "label": "Testing Fundraiser",
      "minter": "0xa513e6e4b8f2a923d98304ec87f64353c4d5c853",
      "numContributors": 17,
      "owner": "0x70997970c51812dc3a010c7d01b50e0d17dc79c8",
      "search": "Testing Fundraiser Testing Token: Fundraising TT4 0x2fc631e4b3018258759c52af169200213e84abab 0xafe1b5bdebd4ae65af2024738bf0735fbb65d44b",
      "softCap": "5000000000",
      "startDate": 1615882704,
      "status": "Running",
      "supply": "100000000000000000000000",
      "token": {
        "id": "0xafe1b5bdebd4ae65af2024738bf0735fbb65d44b"
      },
      "tokenPrice": "0"
    }
  }
}
